costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -444,11 +452,58 @@ void ObstacleLayer::updateBounds(double robot_x, double robot_y, double robot_ya
   updateFootprint(robot_x, robot_y, robot_yaw, min_x, min_y, max_x, max_y);
 }
 
+void ObstacleLayer::computeFootprintCenter(const std::vector<geometry_msgs::Point>& footprint)
+{
+    double sum_x = 0.0, sum_y = 0.0;
+    for (const auto& point : footprint)
+    {
+        sum_x += point.x;
+        sum_y += point.y;
+    }
+
+    footprint_center_.x = sum_x / footprint.size();
+    footprint_center_.y = sum_y / footprint.size();
+}
+
+std::vector<geometry_msgs::Point> ObstacleLayer::getReducedFootprint(const std::vector<geometry_msgs::Point>& original_footprint, double reduction_size)
```

Comment:
reduction_size should be between 0 or 1 right ? Based on you calculations other values would not work as intended. 
Maybe adding some check or static_assert would be useful ? Is var reduction_size are going to be determined during runtime or can it be predicted before execution?
also one more thing... make reduction_size const double&

Reply:
> reduction_size should be between 0 or 1 right ? Based on you calculations other values would not work as intended.
> Maybe adding some check or static_assert would be useful

yes I will add the check to ignore reduction_size if it is not between 0 and 1.

> Is var reduction_size are going to be determined during runtime or can it be predicted before execution?

before execution it has a default value and it can also be changed dynamically as shown in video

> make reduction_size const double&

yes I will change that